### ISSUE TITLE
CDIF (Cross Domain Interoperability Framework)

### ISSUE THREAD

**Issue by nbetancort** [StroemPhi/section-metadata-wg-onto#14]
# Which mapping tool or framework do you want to discuss? 
https://worldfair-project.eu/cross-domain-interoperability-framework/
Overview: https://doi.org/10.5446/66247#t=20:44 or
https://vimeo.com/991198957 (Practical Guidelines for FAIR Interoperability, 2024-07-25)
Deliverable Report: https://doi.org/10.5281/zenodo.11236871

# Why do you think this mapping tool or framework is relevant in the context of our working group?
They work in the context of the WorldFAIR project, that sets out to produce recommendations, interoperability frameworks and guidelines for FAIR data assessment. Their approach is not to create mappings from every existing domain to another, because that is an infinite task, but to create a _lingua franca_ that we can all agree on and communicate with.

# What further steps are needed to be taken or discussed by/in our WG regarding this issue?

0. [x] Start evaluation when more concrete specifications, recommendations or tutorials has been published: 
_Gregory, A. et al. (2024). WorldFAIR (D2.3) Cross-Domain Interoperability Framework (CDIF) (Report Synthesising Recommendations for Disciplines and Cross-Disciplinary Research Areas) (Version 1). Zenodo. https://doi.org/10.5281/zenodo.11236871_
1. [ ] Check and agree internally which parts of CDIF we want to implement and how, as in which areas CDIF can or cannot cover and how deep into details it can be applied.
2. [ ] In case of positive agreement, discuss cooperation ways with the WorldFAIR team (for example, someone from our WG also being part of the CDIF WG)

**Comment by StroemPhi** [comment]
I believe evaluating CDIF is also part of epic #4. 

Although I do agree that the CDIF approach makes total sense, given the fact that it is not yet mature enough, I suggest postponing its evaluation until more concrete specifications, recommendations or tutorials/how-tos are published. Currently, I feel like. "Yes it sounds all nice in theory, but how does it look in reality?"

So the best case would be to have someone from our WG also be part of the CDIF WG, who can regularly brief us on updates.

**Comment by hgoerzig** [comment]
CDIF is a related set of guidelines for providing metadata in a domain-neutral manner. 
 CDIF will provide detailed recommendations for the use of specific standards (Schema.org, DCAT, ODRL, DDI-CDI, SKOS/XKOS, SSSOM, etc.) 
 The usage of the standards is e.g.:
• Understand data structure (DDI-CDI)
• Understand semantics (SKOS/XKOS, OWL, SSSOM)
• Determine origination/context (PROV-O, I-ADOPT/O&M)

It has been applied for the integration of climate data from Copernicus ERA5 and air quality data from the European Environmental Agency (EEA) with data from the European Social Survey (ESS) Integrated Data in ESS Lab. There might be more fields where it has been applied but I don't know about them.

Anyway, the CDIF approach has not been thoroughly enough tested to be recommended. Therefore, volunteers to do so are welcome. This means that it needs to be applied in different science domains. e.g. X-ray Absorption Spectroscopy (XAS) has developed its own de-facto standards, which are not used outside of that community and on different levels of the processing pipeline. Does CDIF work in this area to have a clear mechanism to integrate and use data from different sources with different standards to make XAS data easier discoverable, and to enable exchange it for processing and integration? 

In general we need to check which areas CDIF can or cannot cover and how deep into details it can be applied.

**Comment by nbetancort** [comment]
> Anyway, the CDIF approach has not been thoroughly enough tested to be recommended. Therefore, volunteers to do so are welcome. This means that it needs to be applied in different science domains. e.g. X-ray Absorption Spectroscopy (XAS) has developed its own de-facto standards, which are not used outside of that community and on different levels of the processing pipeline.  Does CDIF work in this area to have a clear mechanism to integrate and use data from different sources with different standards to make XAS data easier discoverable, and to enable exchange it for processing and integration?
> 
> In general we need to check which areas CDIF can or cannot cover and how deep into details it can be applied.

That's very interesting - another example where it could be helpful is in the case of qualitative data (of different types, from different domains).

I think that testing and creating practical guidelines on how to use domain-agnostic standards is very important for interoperability and data exchange purposes, because the use of these powerful and flexible standards (like DDI-CDI, which we are testing for qualitative data https://ddi-alliance.atlassian.net/wiki/spaces/DDI4/pages/3083862017/Qualitative+Data+Subgroup) could be applied in so many ways and for so many use cases that people might need guidelines and support on how to apply them, leading to harmonized use, which in turn leads to interoperability.

So I agree that we need volunteers to bring in different use cases and applications!

**Comment by StroemPhi** [comment]
In the context of our 8-5-2024 call where we talked about using DCAT/-AP, I would be interested in finding out how CDIF is planning on "extending" DCAT to attach domain-specific RDF. (cc @hgoerzig)

**Comment by nbetancort** [comment]
[A webinar took place on Thu 25 July](https://codata.org/the-cross-domain-interoperability-framework-cdif-practical-guidelines-for-fair-interoperability-webinar-25-july/) and covered the following topics:

- Overview of CDIF
- Discovery Profile
- Alignment of CDIF with global digital transformations: A spotlight on the Data and Information Strategy and Implementation Plan of the UN Decade of Ocean Science for Sustainable Development
- Data Description and DDI-CDI
- Data Integration: the SDGs and Google Data Commons
- Future Topics and Engaging with CDIF Development
- Discussion

Recording or presentations:
https://vimeo.com/991198957?share=copy
[Presentations](https://drive.google.com/drive/folders/15sXGZw828n2b-Ana8HYLKVjpPdAnk0ES?usp=share_link)

**Comment by dalito** [comment]
> In the context of our [8-5-2024 call](https://docs.google.com/document/d/1kmAmJ5Uu6zz2A408HE3K0qEvGU2KD5tykLY0iN5nkUc/edit) where we talked about using DCAT/-AP, I would be interested in finding out how CDIF is planning on "extending" DCAT to attach domain-specific RDF.

I did not see an answer to this in the [DCAT chapter of the book](https://cross-domain-interoperability-framework.github.io/cdifbook/metadata/dcat.html). Did they address this in the webinar? I could only attend the first 45 min.

**Comment by nbetancort** [comment]
> I did not see an answer to this in the [DCAT chapter of the book](https://cross-domain-interoperability-framework.github.io/cdifbook/metadata/dcat.html). Did they address this in the webinar? I could only attend the first 45 min.

No, it wasn't addressed, it was more a presentation of general implementation examples and approaches (discovery and integration) while referring to the documentation (for example the book where the profiles are explained and metadata examples are linked: https://cross-domain-interoperability-framework.github.io/cdifbook/examples/index.html and the link to the specific profiles which will be published in the near future https://worldfair-project.eu/cross-domain-interoperability-framework/)

Maybe I got wrong the question from [8-5-2024 call](https://docs.google.com/document/d/1kmAmJ5Uu6zz2A408HE3K0qEvGU2KD5tykLY0iN5nkUc/edit) because I wasn't there at that day, but don't the implementation table and the profiles address that? 

Some notes that I took about more domain-specific information to be addressed by the profiles (in the case of DCAT, rather to facilitate discovery on the web, and not to describe the data in its more granular level - for that the [Data Integration Profile](https://cross-domain-interoperability-framework.github.io/cdifbook/data_integration/dataintegrationintro.html) should be used) from the [examples](https://cross-domain-interoperability-framework.github.io/cdifbook/examples/index.html):
**Geologic Time Scale**
in Abstract says _The Geological Timescale Model is aligned with the W3C OWL-Time ontology https://www.w3.org/TR/owl-time/ for the temporal topology, with OGC GeoSPARQL http://www.opengeospatial.org/standards/geosparql for location data, and with the W3C SOSA/SSN ontology for samples. The content of the vocabulary matches the 2017-02 International Chronostratigraphic Chart_

Then this relationship is made explicit by
```
  dcat:qualifiedRelation [
      rdf:type dcat:Relationship ;
      dcterms:relation <http://resource.geosciml.org/classifier/ics/ischart/> ;
      dcterms:title "OWL ontology defining the boundaries and intervals in the geologic time scale" ;
      dcat:hadRole <http://id.loc.gov/vocabulary/relationship/datasource> ;
    ] ;
```

I would add to it the mention of the W3C OWL-Time ontology and OGC GeoSPARQL from the abstract in a dcterms:conformsTo property

**Comment by StroemPhi** [comment]
I've skimmed through the respective CDIF html book sections and looked at most of the examples. From this, I understand, that currently, we can only use dcat:theme and dcat:keyword to very loosely attach metadata that describes a dataset in terms of its content and we have the DDI-CDI approach limited to CSV. The latter, to me, seems to be a bit more complicated than using some RDF/OWL object property to link to a harmonized content-metadata knowledge graph.

**Comment by nbetancort** [comment]
DDI-CDI can also express wide, long, multidimensional, and key-value pairs, so it deals with the structure of the data, but also with its semantics. So it _should_ be more complicated, but it is also more powerful and allows the expression of domain semantics. This is because the purpose of the standard (and the CDIF functionality it pretends to serve), integration, needs to cover many aspects to make data understandable for its integration into other systems or with other data. I guess they will also provide tools to describe data more easily. We are considering this for non-numerical data as well.

From the book I quote:
_The concept definitions that specify the semantics must be separated from the structural description of the data for a useful cross-domain data description scheme, along with an indication of where the semantics for both the field and the values come from.
CDIF recommends a subset of the classes in the DDI-CDI specification for data description._

This is also where I see the difference with I-Adopt (if I understand I-ADOPT correctly): it only describes the characteristic measured or variable, but not the values, neither structure nor provenance.

DCAT is meant to be used for discovery by CDIF, so I agree with you that the semantics there are loosely handled by the theme and keyword properties.

**Comment by nbetancort** [comment]
In our [2024-07-23 call](https://docs.google.com/document/d/1Dto8TiU5rN2ZmV5dAPbhzapuehYHv8Jtm_Kh0XhVyww/edit) we agreed we should provide use cases from different consortia and present them in CDIF.
There are no templates, but the documentation:
_Gregory, A., Bell, D., Brickley, D., Buttigieg, P. L., Cox, S., Edwards, M., Doug, F., Gonzalez Morales, L. G., Heus, P., Hodson, S., Kanjala, C., Le Franc, Y., Maxwell, L., Molloy, L., Richard, S., Rizzolo, F., Winstanley, P., Wyborn, L., & Burton, A. (2024). WorldFAIR (D2.3) Cross-Domain Interoperability Framework (CDIF) (Report Synthesising Recommendations for Disciplines and Cross-Disciplinary Research Areas) (Version 1). Zenodo. [https://doi.org/10.5281/zenodo.11236871](https://doi.org/10.5281/zenodo.11236871)_

and the [book](https://cross-domain-interoperability-framework.github.io/cdifbook/background/structure.html) (see the examples), that could be followed to describe our use cases.

Since this will probably be done manually, I suggest working on a spreadsheet with the set of fields suggested in each profile?

But first we should discuss in one of our next calls if we should focus on specific profiles (those related to our Mapping/Harmonization topic):

_Users only need adopt those profiles that are useful to them. There is no requirement for the adoption of optional profile content. For example, it is possible to describe data to make it ‘integration ready’ at a detailed level, but not to support profiles for data discovery or access, to give but one example. CDIF profiles are intended to be a toolkit for implementation, with the needed functions being addressed in any specific setting according to implementer priorities._

**Comment by dalito** [comment]
I just checked if IDs resolve https://ddialliance.org/Specification/DDI-CDI/1.0/RDF/Activity/ - but they don't (This is from https://bitbucket.org/ddi-cdi-resources/ddi-cdi/src/master/build/encoding/ontology/Process.onto.ttl). It may just be an oversight but (for me) raises questions on the TRL.

**Comment by nbetancort** [comment]
Not yet officially released

**Comment by dalito** [comment]
Ah, thanks for the info! I may report the issue later in their repo but have no access to my Bibucket-account now.

**Comment by dalito** [comment]
It seems that DDI alliance is slowly moving everything to GitHub. There is also https://ddialliance.github.io/ddimodel-web/ where I could find different representations of the DDI-CDI model. The model is formulated in XMI (UML-exchange format) as primary representation. In a gh-action pipeline it is automatically converted to json-schema and various RDF-outputs (OWL, SHACL, SHEX), see e.g. https://github.com/ddialliance/ddimodel/actions/runs/10402271583 (this link will expire since GH deletes jobs after a while)

The IDs were also resolving when I checked again last week. Progress! :smile:

@nbetancort - I agree with your understanding of I-ADOPT vs. DDI-CDI above. The variable model in I-ADOPT is very flexible which may lead to different decisions on how to represent the same variable. DDI-CDI tries to provide a somewhat tighter framework with their "variable cascade".

**Comment by dalito** [comment]
FYI, slides & notes from the DDI-alliance meeting in Oct-2024 Dagstuhl are available https://ddi-alliance.atlassian.net/wiki/spaces/DDI4/pages/3125674192/2024+Aligning+Technology+Architectures+with+Cross-Domain+Metadata+Models (see also sub-pages of that page)

**Comment by dalito** [comment]
DDI-CDI moved from RC3 state to final (2024-11-22): https://ddialliance.org/announcement/ddi-cdi-version-10-vote-results-approved - However, the Version-1.0 build artifacts are not yet available at the [DDI-CDI specification](https://ddialliance.org/Specification/ddi-cdi) page.

**Comment by matentzn** [comment]
> Do you happen to know if someone already experimented with representing DDI-CDI as LinkML model?

@dalito I think this would be great for a very unrelated issue.. CDIF seems to be pushed by some folks in https://www.rd-alliance.org/groups/fair-mappings-wg/activity/, and I would really love it if somehow I would get some stronger footing for promoting https://linkml.io/linkml-map/ as a way to encode schema cross-walks - which rests a little on the answer to your question...

**Comment by StroemPhi** [comment]
> Should we include LinkML as a mapping tool? (and create a separate issue in this repo for it?) I haven't had a closer look at it yet, and I wonder if it also provides means to map UML classes to RDF classes?

I wouldn't consider it a mapping tool per se. It is only in the regard that you can map the schema entities of it (classes, slots, enums and datatypes) to terminologies. But, to me, this is a totally different kind of mapping, LinkML is rather 

But this is off topic for this issue thread and I would propose to rather use our mailing list or our RockeChat Channel (#section-metadata-wg-onto) for such questions in the future ;) 
(so please don't feel offended, that I will hide your comment for keeping clarity)

**Comment by nbetancort** [comment]
> Not yet officially released

We are pleased to announce that Version 1.0 of DDI Cross-Domain Integration (DDI-CDI) has now been officially published! 

About DDI-CDI

DDI-CDI extends traditional DDI metadata to describe data beyond the social, behavioral, and economic (SBE) domains, addressing the need for broader capabilities. It supports descriptions of event and sensor data ("long" data), key-value data (often associated with "big" data and no-SQL data), and multi-dimensional data. By integrating these with traditional "wide" (or "rectangular") DDI data descriptions, DDI-CDI enables the management and production of integrated data sets from diverse sources.

Additionally, DDI-CDI facilitates the documentation of generic processes, allowing for clear descriptions of the steps involved in data integration. To ensure compatibility with varying technologies and standards used across domains, DDI-CDI is designed to interoperate with a wide range of web standards, supporting both XML- and RDF-based syntax representations.
Overall description of the DDI-CDI product: https://www.ddialliance.org/Specification/DDI-CDI/ 

Read the full DDI-CDI Version 1.0 specification here: https://www.ddialliance.org/Specification/DDI-CDI/1.0/

Learn more: Watch the DDI-CDI overview webinar on YouTube: https://youtu.be/mITmwqcbais
Presentation slides: https://doi.org/10.5281/zenodo.13961159

Issues can be posted here: https://github.com/ddi-cdi/ddi-cdi/issues

**Comment by matentzn** [comment]
@nbetancort I am very interested in the state of Open Data on the who mapping subject, so thank you for sharing DDI-CDI. I have watched all webinars and read some of the docs, but I still don't have a clear picture of the scope of DDI-CDI.

Does the format is supposed to allow any kind of mapping specification (between two arbitrary schemas), or is there always the "DDI" model in the middle? The examples with [SPSS](https://ddialliance.org/hubfs/Specification/DDI-CDI/1.0/example/SPSS_Example.xml) and STATA are a bit opaque for a mere mortal as myself.  For example:
```
<cdi:ValueMapping_formats_DataPoint>
<cdi:ddiReference>
<cdi:dataIdentifier>#dataPoint-0-pplfair</cdi:dataIdentifier>
<cdi:registrationAuthorityIdentifier>int.esseric</cdi:registrationAuthorityIdentifier>
<cdi:versionIdentifier>1</cdi:versionIdentifier>
</cdi:ddiReference>
<cdi:validType>DataPoint</cdi:validType>
</cdi:ValueMapping_formats_DataPoint>
```
I don't understand, for example, how I would express a simple mapping between a `datamodel1:FEMALE` and `datamodelmodel2:F`. Are you aware of any materials for DDI-CDI that could help an outsider with some very simple examples?

A simple example for something like DCAT or schema.org would probably be easier to understand?

**Comment by nbetancort** [comment]
Hi @matentzn Thank you for your question, which I can only answer from a theoretical perspective (as I haven't worked with the model yet) I know that the group is working on a scope document in which they will explain what part this model comes to address in the whole model landscape (together with Schema, org and DCAT, which shouldn't be seen as competitors, but address information at the dataset level, whereas CDI tries to get to the description of each datum - single data instance - in the dataset) I think CDI is more like the I-Adopt model, but much more complex and therefore richer.

As a good introduction I would link to the CDIF documentation, but maybe you have already had a look at it: https://cross-domain-interoperability-framework.github.io/cdifbook/data_integration/ddidescriptiondatastructure.html

At the end of this page your question about ValueMapping is answered (maybe the term mapping is confusing here, as it doesn't mean the kind of mappings our group is dealing with):

_The Logical Record (cdi:LogicalRecord) references all of the Instance variables used within the Data Set. A Physical Segment Layout (cdi:PhysicalSegementLayout) describes the way in which that Logical Record is expressed in the physical file. It has a set of Value Mappings (cdi:valueMapping) instances and corresponding Value Mapping Position (cdi:ValueMappingPosition) instances, which provide the links between the physical layout and the values of the Instance Variables._

From the model itself I found these helpful definitions and examples: https://ddi-cdi.github.io/ddi-cdi_v1.0/field-level-documentation/DDICDILibrary/Classes/FormatDescription/ValueMapping.html

In addition, as you noticed, the core of the CDI model is based on the Variable Cascade concept, which is also at the core of other DDI specifications. However, there will be extensions for other types of data that can't be represented in tabular form (in my case, qualitative data or other non-numerical, non-coded data). For more information on the variable cascade and value domains (the possible set of values that can be used in a variable - it is like constraints or restrictions in the knowledge graph world), I have more information from Dan Gillman, one of the designers of the model, which I can send to you.

In conclusion, I don't think that CDI will be a topic for our WG, but CDIF could be, as it also deals with mappings between terminologies/ontologies. CDI is more about the description and interoperability of the data itself, not the terminologies used to describe it.

**Comment by matentzn** [comment]
@nbetancort Thank you for your detailed explanations! This helped me a lot to contextualise the DDI-CDI approach. My limited intellectual abilities still have trouble understanding what is meant by ValueMapping (which is defined as "Physical characteristics for the value of an instance variable stored in a data point as part of a physical segment layout."), but I can see now that this is a slightly different "concept" of mapping than what the RDA FAIR mappings WG is looking at (note though, we are looking at data mappings as well, not just ontology mappings - just at a higher level, like schema-crosswalks).

**Comment by matentzn** [comment]
Thank you @nbetancort, very thoughful! 

> An example might be the definition of a variable "occupation" in my dataset, where I say (with CDI) that I'm referring to the Schema.org property https://schema.org/hasOccupation (you do this with the attribute [externalDefinition](https://ddi-cdi.github.io/ddi-cdi_v1.0/field-level-documentation/DDICDILibrary/Classes/Conceptual/Concept.html#super-class-hierarchy-generalization:~:text=0..*-,externalDefinition,-%2D%20own%20class) of the concept associated with the variable).

What I would want to see is this:

Given 

```yaml
agent: Nico
occupation: Potter
```

And a target representation

```turtle
my:Nico schemaorg:hasOccupation wikidata:Q3400050
```

How can I express this in CDI-DDI?

**Comment by nbetancort** [comment]
I discussed this with Dan Gillman before replying to you.

I suppose with CDI there won't be a simple representation (or direct path) as it could be in RDF, because the goal of CDI is data integration, and therefore CDI allows representing other things related to the physical representation of semantic information (like the statement above), like that data is stored in some data file, put in context with other variables and semantics, etc. 

If we try to imagine a CDI instance in the form of a graph, the tabular structure where the data is stored in some format/datatype/unit type should also be part of it, which adds complexity to the whole thing, but allows the integration of 2 (or more) datasets with information about the same subject. Nevertheless, the information of your statement "Nico hasOccupation Potter" (in RDF) would be available in a CDI instance and could be visualized/retrieved with the right tools (the CDI WG is also working on this).

CDI instances provide the values in a certain context, in a certain physical representation, with a certain conceptual value from a certain value domain... to be able to take any single datum and merge/integrate it with other datums. Here a snippet of a real example where you can see this more clearly.

[JSON-snippet_CDI](https://github.com/user-attachments/files/18982779/JSON-snippet_CDI.txt)

**Comment by matentzn** [comment]
@nbetancort Thank you! I am beginning to understand now!

**Comment by nbetancort** [comment]
You might find this interesting, it has just been published:

DDI-CDI Working Group. (2025). DDI-CDI and Other Standards (DDI Working Paper Series, No. 35). DDI Alliance. https://doi.org/10.5281/zenodo.11223221

It includes possible uses of I-ADOPT and CDI, and its relationship with CDIF where I highlight the last paragraph:

_CDIF is only in its first published version, and it will be expanded in the future to cover
provenance and data “context”, mapping of semantics and structures in data integration, and
other topics. Additional standards such as PROV, I-ADOPT and SSSOM are seen as likely
candidates for similar inclusion in the recommendations in due course. Thus, the capabilities
for DDI-CDI to combine with other specifications and standards is one which will continue to
prove valuable._

If you are interested in seeing more papers like this with specific topics that could be addressed to make adoption easier in your areas, I can pass the idea on to the CDI and CDIF group.

**Comment by nbetancort** [comment]
The further development of CDIF will continue in the context of EOSC "CDIF4EOSC: Developing and implementing the Cross-Domain Interoperability Framework for EOSC": https://codata.org/initiatives/making-data-work/cdif4eosc/
